recipe/conda_build_config.yaml

```diff
@@ -1,5 +1,5 @@
 python:
```

Comment:
```suggestion
# Workaround: conda-build applies its default `python: {min_pin: x.x, max_pin: x.x}` after it
# resolved `{{ pin_subpackage('python', exact=True) }}`. Override this with "x.x.x.x.x" pins.
pin_run_as_build:
  python:
    min_pin: x.x.x.x.x
    max_pin: x.x.x.x.x
python:
```

Reply:
NB: I did not yet open an issue or draft PR for `conda-build` to resolve this upstream.

Reply:
(We actually need this (or the appropriate fix in `conda-build`) for non-RC builds too -- it just went unnoticed in the release versions.)

Reply:
That doesn't work either. `3.10.0.0.0 > 3.10.0.rc1`. We really need the conda-build PR I linked

Reply:
I tried it locally and it uses the version from the subpackage, i.e., this resolves to `python >=3.10.0rc.1,<3.10.0rc.2.0a0 h49503c6_0_cpython` which is okay-ish.

Reply:
```yaml
$ cat recipe/meta.yaml 
package:
  name: foo
  version: 3.10.0.rc1

outputs:
  - name: foo
  - name: bar
    requirements:
      host:
        - foo
      run:
        - foo
    test:
      commands:
        - echo "hello world"

$ cat recipe/conda_build_config.yaml 
pin_run_as_build:
  foo:
    min_pin: x.x.x.x.x
    max_pin: x.x.x.x.x

foo:
  - 3.10
```

gives me `foo >=3.10,<3.11.0a0`

Reply:
we could just lie about it and call it 3.10.0.dev1.rc1 ?

Reply:
How would that help? `3.10.0.dev1.rc1 < 3.10.0` is still True

Reply:
@isuruf, oh fun. If you use `python` instead of `foo` you get the other behavior I described ("yay" for special case handling -_- ...). (And you get a stack trace if you use `pin_subpackage` with `3.10.0.rc1` instead of `3.10.0rc1`...)